lib/Domain/FilePath.php

```diff
@@ -21,8 +22,26 @@ private function __construct(string $path)
         $this->path = $path;
     }
 
-    public static function fromString(string $path): FilePath
+    public static function fromString(string $string): FilePath
     {
+        $url = parse_url($string);
+
+        if (false === $url) {
+            throw new RuntimeException(sprintf('Cannot guess path from "%s"', $string));
+        }
+
+        $url = $url + ['scheme' => null, 'path' => null];
+
+        ['scheme' => $scheme, 'path' => $path] = $url;
```

Comment:
would it be easier as:

```php
$url = array_merge([
    'scheme' => null,
    'path' => null,
], $url);
```

Reply:
I replaced it with `+=` syntax. Is any other argument for use `array_merge`?

Reply:
I'm not used to the `+` syntax with arrays - so this is done with `array_merge` everywhere else - but practically it seems it can be done as one expression instead of two?

Reply:
I haven't had time to answer, but I hope the current version doesn't raise your doubts. Thanks for merging :)